tendermint/src/private_key.rs

```diff
@@ -38,6 +38,13 @@ impl PrivateKey {
             PrivateKey::Ed25519(signing_key) => Some(signing_key),
         }
     }
+
+    /// Sign a message with this private key
+    pub fn sign(&self, signable_msg_bytes: &[u8]) -> Signature {
+        match self {
+            PrivateKey::Ed25519(signing_key) => signing_key.sign(signable_msg_bytes).into(),
+        }
+    }
```

Comment:
This might be a good application for the [`signature::Signer` trait](https://docs.rs/signature/latest/signature/trait.Signer.html), which would allow `tendermint::PrivateKey` to be used in places which require a generic parameter `S: Signer<Signature>`.